script.R

```diff
@@ -88,61 +130,53 @@ options("repos" = c(
   getOption("repos")
 ))
 
+### install pkg - only if not available
+pkg_name <- read.dcf("DESCRIPTION")[, "Package"]
+if (isFALSE(check_if_pkg_available(pkg_name))) {
+  cli::cli_progress_step("Adding GH release to miniCRAN...")
+  # try to get the package reference from the DESCRIPTION file (URL field)
+  pkg_url <- gsub("\n|/$", "", strsplit(read.dcf("DESCRIPTION")[1, "URL"], ",")[[1]])
+  pkg_url_gh <- grep("github.com", pkg_url, value = TRUE)
+  pkg_ref_released <- paste0(gsub(".*github.com/", "", pkg_url_gh), "@*release")
+  if (length(pkg_ref_released) == 0) {
+    cli::cli_abort("Unable to automatically determine the package reference.")
+    return(NULL)
+  }
+  install_and_add_to_minicran(pkg_ref_released, minicran_path)
+}
+
+### add pkgs from GH to miniCRAN
```

Comment:
Why do we need `miniCRAN` at all?

Reply:
revdepcheck uses the install from `crancache` (unforunately not pak install which is able to pick up from its cache)
`crancache::install()` is a default install with its own cache

At first I was trying to put something in the cache but I failed.
Another idea was to extend the repos my miniCRAN so that `crancache::install()` can find those packages. So what I'm doing here is to pak::install, take it from cache and add to the minicran repo.

Reply:
Alrighty, I think it would be good to add a bit explanation to the whole flow, as a comment on top of the document, so we remember in 6 months!

Reply:
OK I will add some comments to my local clone. If I would change this next time - I will include this as well

Reply:
https://github.com/insightsengineering/r-revdepcheck-action/pull/5